Source/background.js

```diff
@@ -76,31 +60,111 @@ function translateOneRequestPerFewWords(words, prefs, callback) {
     translateORPFWRec(concatWordsArray, 1, cWALength, tMap, prefs, callback);
 }
 
+var translator = {
+    translate: function(prefs, word) {}
+};
+
+var googleTranslator = function() {};
+googleTranslator.prototype = Object.create(translator);
+
+var yandexTranslator = function() {};
+yandexTranslator.prototype = Object.create(translator);
+
+/**
+ * @desc Constructs google translate url
+ * @params User saved preferences
+ * @params word to be translated
+ * @return Google Translate query from source to destination language
+ */
+googleTranslator.prototype.translate = function(prefs, word, callback){
+    var url = 'http://translate.google.com/translate_a/t?client=f&otf=1&pc=0&hl=en';
+    var sL = prefs["sourceLanguage"];
+    if(sL != 'auto') {
+        url += '&sl=' + sL;
+    }
+    url += '&tl=' + prefs["targetLanguage"];
+    url += '&text=';
+    url += word;
+
+    getData(url, function(result){
+        var tMap = {}
+        for (i in result.sentences) {
+            var orig = result.sentences[i].orig;
+            var origT = orig.substring(0,orig.length - 1);
+            var trans = result.sentences[i].trans;
+            var transT = trans.replace(/[.\u3002]/,""); // removes punctuation
+            tMap[origT] = transT;
+        }
+        callback(tMap);
+    });
+}
+
+yandexTranslator.prototype.translate = function(prefs, word, callback){
+    var apikey = "trnsl.1.1.20150429T202526Z.d63a1483e719555a.852f432bad29a4628c1d1694b6114c0efb22f183"
```

Comment:
Besides the limit of 1000000 characters per day, there is also a limit of 10000000 characters per month.

MindTheWord currently has 10000 users. This implies an average of only 100 characters per user per day, if we go with a single key for all users. This would lead very quickly to all users being blocked. Therefore, I think we must go with one key per user, and ask users to generate their keys.

How is your time availability now? Do you think you have the time to add a field in the settings, where the user could specify his/her key?